lib/streamlit/file_util.py

```diff
@@ -124,6 +124,44 @@ def get_assets_dir():
     return os.path.normpath(os.path.join(dirname, "static/assets"))
 
 
+# TODO Rename function to more meaningful name
+def sanitize_user_static_path(static_folder_path):
+    main_script_directory = os.getcwd()
+    real_path = os.path.realpath(static_folder_path)
+
+    pages_directory = os.path.abspath("pages")
+    config_directory = os.path.abspath(CONFIG_FOLDER_NAME)
+
+    if (
+        os.path.commonprefix([real_path, main_script_directory])
+        != main_script_directory
+    ):
+        raise RuntimeError(
+            f"Static file directory {real_path} is served outside of "
+            f"Streamlit project root."
+        )
+
+    # TODO [KAREN] Add comment about symlink
+    if real_path == main_script_directory:
+        raise RuntimeError(
+            "For security reasons we dont allow expose whole project directory"
+        )
+
+    if os.path.commonpath([pages_directory, real_path]) == pages_directory:
+        raise RuntimeError(
+            f"Static file dir couldn't be 'pages' or inside 'pages' "
+            f"directory. Current  specified directory {real_path}"
+        )
+
+    if os.path.commonpath([config_directory, real_path]) == config_directory:
+        raise RuntimeError(
+            "STATIC FOLDER COULD NOT BE .streamlit or inside .streamlit because of "
+            "security, this could lead to secrets and configs exposure."
```

Comment:
`For security reasons we don't enable static serving from './.streamlit' since it may contain secrets and configuration.`